string six_face_str
---
string solve_str
